server/services/operations_service.go

```diff
@@ -125,18 +128,70 @@ func searchFilterFunc(query string) func(b *pb.Blurb) bool {
 	}
 }
 
+// returns a successful response if the resource name is not blank
 func (s operationsServerImpl) CancelOperation(ctx context.Context, in *lropb.CancelOperationRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "google.longrunning.CancelOperation is unimplemented.")
+	if in.Name == "" {
+		return nil, status.Error(codes.NotFound, "cannot cancel operation without a name.")
+	}
+	return &empty.Empty{}, nil
 }
 
+// returns a fixed response matching the given PageSize if the resource name is not blank
 func (s operationsServerImpl) ListOperations(ctx context.Context, in *lropb.ListOperationsRequest) (*lropb.ListOperationsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "google.longrunning.ListOperations is unimplemented.")
+	if in.Name == "" {
+		return nil, status.Error(codes.NotFound, "cannot list operation without a name.")
+	}
+
+	var operations []*lropb.Operation
+	if in.PageSize > 0 {
+		for i := 1; i <= int(in.PageSize); i++ {
+			var result *lropb.Operation_Response
+			if i%2 == 0 {
+				result = &lropb.Operation_Response{}
+			}
+			operations = append(operations, &lropb.Operation{
+				Name:   "the/thing/" + strconv.Itoa(i),
+				Done:   result != nil,
+				Result: result,
+			})
+		}
+	} else {
+		operations = append(operations, &lropb.Operation{
+			Name: "a/pending/thing",
+			Done: false,
+		})
+	}
+
+	return &lropb.ListOperationsResponse{
+		Operations:    operations,
+		NextPageToken: "txenNext",
+	}, nil
 }
 
+// returns a successful response if the resource name is not blank
 func (s operationsServerImpl) DeleteOperation(ctx context.Context, in *lropb.DeleteOperationRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "google.longrunning.DeleteOperation is unimplemented.")
+	if in.Name == "" {
+		return nil, status.Error(codes.NotFound, "cannot delete operation without a name.")
+	}
+	return &empty.Empty{}, nil
 }
 
+// randomly waits and returns an operation with the same name
 func (s operationsServerImpl) WaitOperation(ctx context.Context, in *lropb.WaitOperationRequest) (*lropb.Operation, error) {
-	return nil, status.Error(codes.Unimplemented, "google.longrunning.WaitOperation is unimplemented.")
+	if in.Name == "" {
+		return nil, status.Error(codes.NotFound, "cannot wait on a operation without a name.")
+	}
+
+	num := rand.Intn(500)
+	time.Sleep(time.Duration(num) * time.Millisecond)
```

Comment:
Why not use the timeout in [lropb.WaitRequest](https://github.com/googleapis/go-genproto/blob/fa694d86fc64c7654a660f8908de4e879866748d/googleapis/longrunning/operations.pb.go#L400)? The page size was used in the List RPC. I have a feeling folks would expect the timeout to be respected. WDYT?

Reply:
I don't really have a preference. The [`WaitOperation`](https://github.com/googleapis/googleapis/blob/2a2c5518f64010c4e458afc818e57ed24fecdf6d/google/longrunning/operations.proto#L116) is a little strange anyway. The docs basically say that it's free to ignore the timeout and it doesn't even have to wait. So, random seemed fitting to me.

Reply:
Ok fine by me